src/lib/Scenes/Fair/Fair.tsx

```diff
@@ -17,40 +20,48 @@ interface Props extends ViewProperties {
 export class Fair extends React.Component<Props> {
   navigator?: NavigatorIOS
 
-  handleViewAllExhibitorsPressed = () => {
+  navigate = ({ component, props }: { component: any; props?: object }) => {
     if (!this.navigator) {
       throw new Error("navigator is undefined")
     }
-
     this.navigator.push({
-      component: FairExhibitorsScreen,
+      component,
+      passProps: {
+        ...this.props,
+        onViewAllExhibitorsPressed: this.handleViewAllExhibitorsPressed,
+        onViewFairArtworksPressed: this.handleViewFairArtworksPressed,
+        onViewFairArtistsPressed: this.handleViewFairArtistsPressed,
+        onViewFairBoothPressed: this.handleViewFairBoothPressed,
+        onViewFairBoothArtworksPressed: this.handleViewFairBoothArtworksPressed,
+        onViewFairBoothArtistsPressed: this.handleViewFairBoothArtistsPressed,
+        ...(props || {}),
+      },
       title: "",
-      passProps: this.props,
     })
   }
 
-  handleViewAllArtworksPressed = () => {
-    if (!this.navigator) {
-      throw new Error("navigator is undefined")
-    }
+  handleViewAllExhibitorsPressed = () => {
+    this.navigate({ component: FairExhibitorsScreen })
+  }
 
-    this.navigator.push({
-      component: FairArtworksScreen,
-      title: "",
-      passProps: this.props,
-    })
+  handleViewFairArtworksPressed = () => {
+    this.navigate({ component: FairArtworksScreen })
   }
 
-  handleBrowseArtistsPressed = () => {
-    if (!this.navigator) {
-      throw new Error("navigator is undefined")
-    }
+  handleViewFairArtistsPressed = () => {
+    this.navigate({ component: FairArtistsScreen })
+  }
 
-    this.navigator.push({
-      component: FairArtistsScreen,
-      title: "",
-      passProps: this.props,
-    })
+  handleViewFairBoothPressed = props => {
+    this.navigate({ component: FairBoothScreen, props })
+  }
+
+  handleViewFairBoothArtistsPressed = props => {
+    this.navigate({ component: ShowArtistsScreen, props })
```

Comment:
This props passing is kind of awkward but necessary with the current layout as far as I can tell - We let the root view handle navigation, but the root view is missing the data required to show screens of other types. In this case, `handleViewFairBoothPressed`, `handleViewFairBoothArtistsPressed`, and `handeleViewFairBoothArtworksPressed` all receive fragments on `Show` rather than `Fair` - at what point would it make sense to split this out into a separate query renderer?

Reply:
I would argue it makes sense to have the `FairBoothScreen` and `ShowArtworksScreen` into their own QueryRenderers for the reason you mentioned but also because currently we're overfetching data when rendering the base Fair screen since it doesn't need artists or exhibitors yet. But I haven't thought through all the repercussions of switching to nested QueryRenderers so let's keep going that route for now